source/common/redis/proxy_filter.cc

```diff
@@ -19,9 +19,15 @@ ProxyFilterConfig::ProxyFilterConfig(const Json::Object& config, Upstream::Clust
       cluster_name_(config.getString("cluster_name")),
       stat_prefix_(fmt::format("redis.{}.", config.getString("stat_prefix"))),
       stats_(generateStats(stat_prefix_, scope)) {
-  if (!cm.get(cluster_name_)) {
-    throw EnvoyException(
-        fmt::format("redis filter config: unknown cluster name '{}'", cluster_name_));
+  Upstream::ThreadLocalCluster* cluster = cm.get(cluster_name_);
+  if (cluster == nullptr) {
+    throw EnvoyException(fmt::format("redis filter config: unknown cluster '{}'", cluster_name_));
+  }
+
+  if (cluster->info()->addedViaApi()) {
```

Comment:
I went back and forth on this. I agree it's a lot of churn. On the other hand, I think it does have value in that I was planning on making sure that the RDS cluster for example references a static cluster that can't be removed. I also think that giving the people the ability to write filters that don't work with CDS is probably OK as long as they have error/config checking. So, I think I'm still leaning towards adding it, but I could be swayed either way.